### ISSUE TITLE
//fixwater is only filling a top layer of water

### ISSUE THREAD

**Issue by LadyCailin** [LadyCailin/TestIssueUploader#120]
WORLDEDIT-2206 - Reported by sk89q on 2012-10-29 18:55:20 UTC

Due to recent changes in how WorldEdit sets blocks, only the top layer of water is set (as stationary water) and it doesn't fill the empty blocks below.

**Comment by LadyCailin** [comment]
Comment by Puremin0rez on 2012-11-03 11:00:58 UTC

Horribly annoying issue! Hope it's fixed soon.

**Comment by LadyCailin** [comment]
Comment by Michael.Schäffer on 2012-11-05 01:55:57 UTC

2 Layers of water are set. they are not flowing down to fill the space.

workaround is for example: //fillr water 20 20

**Comment by LadyCailin** [comment]
Comment by bobmasterb on 2012-12-21 11:22:23 UTC

SO ANNOYING!

**Comment by LadyCailin** [comment]
Comment by Dylan.Colborne on 2012-12-23 00:20:44 UTC

I think it is because the command fills in the air with block 9 (stagnant water) which doesn't move until it is updated.

**Comment by LadyCailin** [comment]
Comment by TheStriker095 on 2013-01-13 10:04:06 UTC

Bump...

**Comment by LadyCailin** [comment]
Comment by wizjany on 2013-01-13 10:28:37 UTC

fixwater is only supposed to fix flowing water where it should be stationary
as the comment above says, large hole should be fixed via other methods such as fill

**Comment by LadyCailin** [comment]
Comment by DHLF on 2013-03-04 19:04:52 UTC

Many times I have to drain some lakes. 
Sadly now its not easy to put those water back into those lakes after the *//drain* command as now *//fixwater* changed somehow.

To get back to the prior behavior of *//fixwater* you now have to
*1.) Using a bucket of water and put water at the ground
*2.) Stand near those waterblock and use *//fixwater 10* (keep the radius as low as possible for testing purposes) 
 so WE will fill two layers of static water keeping the lower levels.. empty!
*3.) select the surface of the new water (and the second layer) and use *//replace 9 8*
 This way those very static water will be replaced with *normal* flooding water.
 So the water will fall down without flooding whole caves as it would be done by *//fillr water 10*

I am not happy with the new behaviour of this command but it is now faster(as not so many blocks are changed).
Maybe the command could be split like *//fixwater* (the old fashioned way which uses block code 8)
and a new version like *//fixxwater -s* (for static water block code 9)?

**Comment by LadyCailin** [comment]
Comment by wizjany on 2013-03-04 19:23:11 UTC

drain command changed?
I am not exactly sure why all these commands are changing all of a sudden...I don't use them often myself, but I certainly don't remember them being changed...

**Comment by LadyCailin** [comment]
Comment by sk89q on 2013-03-04 20:31:10 UTC

No, this is a bug and it changed. It makes //fixwater practically pointless.

y u close a ticket I made

hater

**Comment by LadyCailin** [comment]
Comment by DHLF on 2013-03-05 19:51:30 UTC

Sorry, for not having clearly written what I wanted to say.
I corrected my last comment. Some of the text was missing.
The */drain* command did not have been changed.
So I have misunderstood the idea behind */fixwater* as I used it often to change broken water at surfaces and to fill drained lakes :)
Nevertheless those workaround will do, thank you prog and wizjany for those hint.
